File: src/main/scala/com/zy/commonapi/MyPartitioner.scala
package com.zy.commonapi

import org.apache.flink.api.common.functions.Partitioner

class MyPartitioner extends Partitioner[Long]{
  override def partition(k: Long, numPartitions: Int): Int = {
    println("分区总数：" + numPartitions)
    if(k%2==0){
      0
    }else{
      1
    }
  }
}

File: src/main/java/com/zy/comapi/MyPartitioner.java
package com.zy.comapi;

import org.apache.flink.api.common.functions.Partitioner;

public class MyPartitioner implements Partitioner<Long> {
    @Override
    public int partition(Long key, int numPartitions) {
        if(key % 2 ==0){
            return 0;
        }else{
            return 1;
        }
    }
}
